riscv_analysis/src/cfg/iterator.rs

```diff
@@ -0,0 +1,88 @@
+use crate::cfg::{Cfg, CfgNode};
+use std::{collections::HashSet, rc::Rc};
+
+/// Iterate over a CFG using the underlying node order.
+///
+/// This iterator is useful when you don't care about the order you iterate over
+/// the nodes.
+pub struct CfgIterator {
```

Comment:
My reason for copying is that it may be reasonable for some passes to modify the CFG. Do you think that we will have passes that will modify the CFG?

Reply:
It is definitely possible to have some passes modify a CFG, and we should worry about that case as well.

If this is the case, my thinking was that the user of the iterator would call something like `cfg.clone().iter()`, so that it gives the choice to the caller whether they want to clone or not. As well, if the `CfgIterator` contained a read-only reference to `Cfg`, the borrow checker *should* stop anyone from modifying anything in `cfg` if it is iterating on `cfg.iter()`.

Reply:
That makes sense